app/models/course_data/assignment.rb

```diff
@@ -65,6 +68,14 @@ def sibling_assignments
       .where.not(user: user_id)
   end
 
+  def status
+    AssignmentPipeline.new(assignment: self).status
```

Comment:
Better to just inflate one `AssignmentPipeline` per `Assignment`, I think, and re-use it across methods.

Reply:
What do you mean by "inflate?" Something like the following in each?
```ruby
def status
  @pipeline ||= AssignmentPipeline.new(assignment: self)
  @pipeline.status
end
```